app/src/main/java/org/sopt/sample/fragments/adapters/GithubUsersAdapter.kt

```diff
@@ -0,0 +1,72 @@
+package org.sopt.sample.fragments.adapters
+
+import android.content.Context
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.NonDisposableHandle.parent
+import org.sopt.sample.databinding.ItemDescriptionBinding
+import org.sopt.sample.fragments.models.UserRepo
+import org.sopt.sample.databinding.ItemUsersBinding
+
+class GithubUsersAdapter(context: Context):RecyclerView.Adapter<RecyclerView.ViewHolder>(){
+    private val inflater by lazy { LayoutInflater.from(context) }
+    private var userList: List<UserRepo> = emptyList()
+
+    override fun getItemViewType(position: Int): Int {
+        return userList[position].dataType
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when(viewType) {
+            type1 -> {
+                val binding = ItemDescriptionBinding.inflate(inflater, parent, false)
+                DescriptionViewHolder(binding)
+            }
+            else -> {
+                val binding = ItemUsersBinding.inflate(inflater, parent, false)
+                GithubUsersViewHolder(binding)
+            }
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when(userList[position].dataType){
+            type1 -> {
+                holder as DescriptionViewHolder
+                holder.onBind(userList[position])
+            }
+            else -> {
+                holder as GithubUsersViewHolder
+                holder.onBind(userList[position])
+            }
+        }
+    }
+
+    override fun getItemCount(): Int {
+        return userList.size
+    }
+
+    fun setUserList(userList: List<UserRepo>){
+        this.userList = userList.toList() // toList 해주는 이유 > 얕은 복사를 위해
+        notifyDataSetChanged()
+    }
+
+    inner class GithubUsersViewHolder(private val binding: ItemUsersBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(data : UserRepo){
+            binding.ivGithubLogo.setImageResource(data.image!!)
+            binding.tvRepoName.text = data.repositoryName
+            binding.tvUserName.text = data.name
+        }
+    }
+    inner class DescriptionViewHolder(private val binding: ItemDescriptionBinding) : RecyclerView.ViewHolder(binding.root) {
+        fun onBind(data : UserRepo){
+            binding.tvRepoDescription.text = data.name
+        }
+    }
```

Comment:
```suggestion
    class GithubUsersViewHolder(private val binding: ItemUsersBinding) : RecyclerView.ViewHolder(binding.root) {
        fun onBind(data : UserRepo){
            binding.ivGithubLogo.setImageResource(data.image!!)
            binding.tvRepoName.text = data.repositoryName
            binding.tvUserName.text = data.name
        }
    }
    class DescriptionViewHolder(private val binding: ItemDescriptionBinding) : RecyclerView.ViewHolder(binding.root) {
        fun onBind(data : UserRepo){
            binding.tvRepoDescription.text = data.name
        }
    }
```
만약, Inner class에서 Outer class 정보를 가지고 있고, 그 참조를 해지하지 못하게 되면 메모리 누수가 생길 수도 있습니다.  
따라서, inner class를 사용하지않고, nested class를 사용하는 것을 권장합니다 😄

Reply:
오 inner 키워드를 사용안하면 자동으로 중첩 클래스(nested class)가 되는거군요 !!
메모리 누수에 관한 내용도 
 https://shinjekim.github.io/kotlin/2019/08/29/Kotlin-%EB%82%B4%EB%B6%80-%ED%81%B4%EB%9E%98%EC%8A%A4(inner-class)%EC%99%80-%EC%A4%91%EC%B2%A9-%ED%81%B4%EB%9E%98%EC%8A%A4(nested-class)/ 
요기서 확인했습니다 감사합니다 !!